metrics/sinks/hawkular/driver_historical.go

```diff
@@ -0,0 +1,432 @@
+
+				for _, bdp := range bdps {
+					ta, err := h.bucketPointToAggregationValue(bdp, &typ, aggregations, bucketSize)
+					if err != nil {
+						errChan <- err
+						return
+					}
+
+					tav = append(tav, ta)
+				}
+				resLock.Lock()
+				res[key] = tav
+				resLock.Unlock()
+			}
+
+		}(keyM)
+	}
+	wg.Wait()
+	close(errChan)
+
+	// Check for errors first
+	for e := range errChan {
+		return nil, e
+	}
+
+	return res, nil
+}
+
+// GetMetricNames retrieves the available metric names for the given object
+func (h *hawkularSink) GetMetricNames(metricKey core.HistoricalKey) ([]string, error) {
+	tags := h.keyToTags(&metricKey)
+	tags[descriptorTag] = "*"
+
+	var r map[string][]string
+	var err error
+
+	if h.isNamespaceTenant() && metricKey.NamespaceName != "" {
+		r, err = h.client.TagValues(tags, metrics.Tenant(metricKey.NamespaceName))
+	} else {
+		r, err = h.client.TagValues(tags)
+	}
+
+	if r != nil {
+		return r[descriptorTag], err
+	}
+	return nil, err
+}
+
+// GetNodes retrieves the list of nodes in the cluster
+func (h *hawkularSink) GetNodes() ([]string, error) {
+	tags := make(map[string]string)
+	tags[core.LabelHostname.Key] = "*"
+	tags[core.LabelMetricSetType.Key] = core.MetricSetTypeNode
+	r, err := h.client.TagValues(tags) // Assume these are stored in the system tenant
+	if r != nil {
+		return r[core.LabelHostname.Key], err
+	}
+	return nil, err
+}
+
+// GetPodsFromNamespace retrieves the list of pods in a given namespace
+func (h *hawkularSink) GetPodsFromNamespace(namespace string) ([]string, error) {
+	tags := make(map[string]string)
+	tags[core.LabelPodName.Key] = "*"
+	tags[core.LabelMetricSetType.Key] = core.MetricSetTypePod
+
+	var r map[string][]string
+	var err error
+
+	if h.isNamespaceTenant() {
+		r, err = h.client.TagValues(tags, metrics.Tenant(namespace))
+	} else {
+		tags[core.LabelNamespaceName.Key] = namespace
+		r, err = h.client.TagValues(tags)
+	}
+
+	if r != nil {
+		return r[core.LabelPodName.Key], err
+	}
+	return nil, err
+}
+
+// GetSystemContainersFromNode retrieves the list of free containers for a given node
+func (h *hawkularSink) GetSystemContainersFromNode(node string) ([]string, error) {
+	tags := make(map[string]string)
+	tags[core.LabelContainerName.Key] = "*"
+	tags[core.LabelHostname.Key] = node
+	tags[core.LabelMetricSetType.Key] = core.MetricSetTypeSystemContainer
+	r, err := h.client.TagValues(tags) // I assume these are stored at the system tenant
+	if r != nil {
+		return r[core.LabelContainerName.Key], err
+	}
+	return nil, err
+}
+
+// GetNamespaces retrieves the list of namespaces in the cluster
+func (h *hawkularSink) GetNamespaces() ([]string, error) {
+	// Fetch tenants, if the labelToTenantId is set (and includes namespace label)
+	if h.isNamespaceTenant() {
+		tds, err := h.client.Tenants()
+		if err != nil {
+			return nil, err
+		}
+		ns := make([]string, 0, len(tds))
+		for _, td := range tds {
+			ns = append(ns, td.ID)
```

Comment:
Currently such setup is not supported and we do not have any metadata associated to filter out non-Heapster tenants at this point.